src/test/java/io/reactivex/single/SingleTest.java

```diff
@@ -188,7 +188,10 @@ public void testFlatMap() {
             }
         }
         ).subscribe(ts);
-        ts.awaitTerminalEvent();
+        if (!ts.awaitTerminalEvent(5, TimeUnit.SECONDS)) {
```

Comment:
you can also use `assertTrue(ts.await...)` for handy failures when it doesn't time out